server/channel.py

```diff
@@ -1,2 +1,99 @@
+import json
+
 class NoSuchChannelError(Exception):
     pass
+
+class ChannelCreationError(Exception):
+    pass
+
+class Channel(object):
+
+    @classmethod
+    def create(cls):
+        """
+        Creates a new channel.
+        """
+        pass
+
+    def __init__(self, name, broadcasterUUID):
+        self.name = name
+        self.broadcaster = broadcasterUUID
+        self.userlist = []
+        self.url = "http://baseurl.com/" + name
+        self.add_listener(name)
+        super(Channel, self).__init__()
+
+    num_listeners = property(lambda self: len(self.userlist))
+
+    def add_listener(self, listener):
+        if not (listener in self.userlist):
+            self.userlist.append(listener)
+            return True
+        else:
+            return False
+
+    def remove_listener(self, listener):
+        if not (listener in self.userlist):
+            return False
+        else:
+            self.userlist.remove(listener)
+            return True
+
+    def is_in_channel(self, name):
+        return (name in self.userlist)
+
+    def is_same_uuid(self, id):
+        return (self.broadcaster == id)
+
+    def create_metadata(self):
+        with open('./metadata.txt','w') as fin:
```

Comment:
fyi relative paths like this are very error prone, always use ```os.getcwd()``` instead of the dot when openning files.